crates/core/src/models/allowance.rs

```diff
@@ -0,0 +1,48 @@
+use std::marker::PhantomData;
```

Comment:
I think there as been a bit of a misunderstanding as to why we implemented the `TokenBalances` in the way we did, using `FutureTokenBalance` and implementing `Future for FutureTokenBalance`: the main reason was that we needed a way to return futures in a loop, which could then be joined with `join_all` and resolved asynchronously, all the while linking the balance we got to the address of the contract we got this balance from. You can see all this [here](https://github.com/kkrt-labs/kakarot-rpc/blob/main/crates/core/src/client/mod.rs#L389). However, in your case, you are just calling `token_allowance`, retrieving the allowance for one owner, one spender and one contract. There is no need to implement all the future resolution handling.

Please let me know if this is clear and otherwise DM me on telegram.